### PR TITLE
Charlesmchen/message view perf2

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk 

See: https://github.com/WhisperSystems/SignalServiceKit/pull/228

### CODE REVIEW

File: Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m [hunk 1]
```diff
@@ -224,7 +224,11 @@ - (UIView *)createVideoMediaView
 
     CGSize size = [self mediaViewDisplaySize];
 
-    UIImageView *imageView = [[UIImageView alloc] initWithImage:self.image];
+    UIImage *image = self.attachment.image;
+    if (!image) {
+        return nil;
```
Comment (michaelkirk) [inline Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m]: Should this be logged? asserted?

File: Signal/src/ViewControllers/MessagesViewController.m [hunk 2]
```diff
@@ -758,11 +758,11 @@ - (void)configureForThread:(TSThread *)thread
         [[YapDatabaseViewMappings alloc] initWithGroups:@[ thread.uniqueId ] view:TSMessageDatabaseViewExtensionName];
     [self.uiDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
       [self.messageMappings updateWithTransaction:transaction];
-      self.page = 0;
-      [self updateRangeOptionsForPage:self.page];
-      [self.collectionView reloadData];
     }];
+    self.page = 0;
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/MessagesViewController.m]: Hmm... this seems like a reasonable way to maybe avoid some kind of layout race condition.  👍 

Did you observe a change, or is this a hunch?

Comment (charlesmchen) [inline Signal/src/ViewControllers/MessagesViewController.m]: Oh, my intention was simply to move unnecessary work out of the transaction in order to keep transaction lifetime as short as possible.

File: Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m]: Done.

**Pull request merged**